src/core/com/channel.h

```diff
@@ -0,0 +1,123 @@
+/*******************************************************************************
+ * Copyright (c) 2025 Martin Erich Jobst
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0.
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *    Martin Jobst - initial implementation
+ *******************************************************************************/
+
+#pragma once
+
+#include <span>
+#include <string>
+#include <utility>
+
+#include "common.h"
+#include "observer.h"
+
+namespace forte::com {
+
+  template<typename T>
+  class ComChannel {
+    public:
+      /**
+       * \brief Open a channel with the given address and descriptors
+       * \param paAddress The address to open the channel with
+       * \param paDescriptors Communication channel descriptors for configuration
+       * \return The result of opening the channel
+       * \retval Ok The channel was opened successfully
+       * \retval InvalidId The address or descriptors are invalid
+       * \retval Overflow There are not enough resources
+       */
+      virtual ComResult open(std::string_view paAddress, std::span<ComChannelDescriptor> paDescriptors) = 0;
```

Comment:
is the name address not a bit limiting. How I understand the setup it is similar to the existing communication infrastructure all configuration data for a ComChannel or? Would then something like `paChannelConfData` be more appropriate?